### PR TITLE
Fix large pages on Windows and make them more user-friendly

### LINKED ISSUE #4: active self test
All old miner versions contains a self test of the cpu miner. The code is currently not called and returned always true.

# Todo

- [ ] Activate and test the *self test* functionality.

### PR DESCRIPTION
**Pull request by fireice-uk**
- Add large page initialisation on Windows
- Add a proper UAC manifest requesting elevation when we start
- Fix #4 self_test() - this is integral part of initialisation

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -401,6 +401,10 @@ set(LIBRARY_OUTPUT_PATH "bin")
 
 target_link_libraries(xmr-stak ${LIBS} xmr-stak-c xmr-stak-backend)
 
+if(CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
+    set_property(TARGET xmr-stak PROPERTY LINK_FLAGS "/level='requireAdministrator' /uiAccess='false'")
```
Comment (psychocrypt) [inline CMakeLists.txt]: Does this mean that the miner must run always started as administrator under windows?

**Comment by fireice-uk** [comment]
@psychocrypt I hacked around UAC in a different way. Now the miner will run no matter if the user presses yes or no, however if he presses yes it will run in a different process than original.

Middle ground is impossible here because UAC would be too easy to bypass if you could elevate a running process.

**Comment by psychocrypt** [comment]
Is it really needed that the UAC always pops up? Or do wee need it only
once and than large pages is activated on the system.
In the case we need it only one we should think about to it only if no
config.txt is available or build a extra little helper tool which can but
not must executed by the user to activate large pages.

Am 05.10.2017 12:17 Vorm. schrieb "fireice-uk" <notifications@github.com>:

@psychocrypt <https://github.com/psychocrypt> I hacked around UAC in a
different way. Now the miner will run no matter if the user presses yes or
no, however if he presses yes it will run in a different process than
original.

Middle ground is impossible here because UAC would be too easy to bypass if
you could elevate a running process.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/fireice-uk/xmr-stak/pull/12#issuecomment-334305553>,
or mute
the thread

**Comment by fireice-uk** [comment]
@psychocrypt UAC prompt is needed if you want to allocate large pages, so every time unfortunately, unless the uses only wants to use slow memory.

**Comment by psychocrypt** [comment]
:-( I will try you changes tomorrow, now it's to late

**Pull request merged**